lib/crc32.c

```diff
@@ -0,0 +1,100 @@
+/*

```

Comment:
Similar to the header, I don't see why we can't just put this in a common crc file.